tests/mean_04.cc

```diff
@@ -0,0 +1,125 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2019 by the SampleFlow authors.
+//
+// This file is part of the SampleFlow library.
+//
+// The SampleFlow library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of SampleFlow.
+//
+// ---------------------------------------------------------------------
+
+
+// Check the MeanValue consumer
+
+
+#include <iostream>
+#include <valarray>
+#include <random>
+
+#include <sampleflow/producers/metropolis_hastings.h>
+#include <sampleflow/consumers/mean_value.h>
+
+
+class MyTriangle
+
+{
+  public:
+
+    std::valarray<double> side_lengths;
```

Comment:
You actually know the number of elements you need to store. So
```suggestion
    std::array<double,3> side_lengths;
```
seems like a better data type.

Reply:
Ah nice, this did not occur to me.